Mta item page

MarkAlperin opened the pull request:
now has functionality for claim button put request

MarkAlperin committed:
merge from ian
Merge branch 'ian' of https://github.com/GreeneOcean/Project_Greene into ian

MarkAlperin committed:
two map options functional, choosing between them

MarkAlperin committed:
Pins are rendered, working on Items/List and ItemCard

MarkAlperin committed:
list and map functional and linked

MarkAlperin committed:
implemented filter functionality

MarkAlperin committed:
added button to selected item for page navigation, also fixed bug with distance filter and removed some comments/logs/extraneous code

MarkAlperin committed:
add package.json changes

MarkAlperin committed:
big merge of browsePage
Merge branch 'dev' of https://github.com/GreeneOcean/Project_Greene into mta_browse_page_mvp_functionality

MarkAlperin committed:
final touches for merge

MarkAlperin committed:
deleted bundle.LISCENESE

MarkAlperin committed:
made ProductList and ProductCard for HomeItemsWidget

MarkAlperin committed:
completed dummyData dummyFunctionality  widget (no state, no item click functionality)

MarkAlperin committed:
text display acceptable with dummyData

MarkAlperin committed:
pulling from mta_browse_page_mvp_functionality to make requested changes
Merge branch 'mta_browse_page_mvp_functionality' of https://github.com/GreeneOcean/Project_Greene into mta_browse_page_mvp_functionality

MarkAlperin committed:
make BrowsePage container name more descriptive

MarkAlperin committed:
removed process.env import

MarkAlperin committed:
changed google map api key from process.env to a temp string placeholder

MarkAlperin committed:
removed comment from MapMarker.js

MarkAlperin committed:
removed old fetch request, need to replace later

MarkAlperin committed:
replace styled-component names with more descriptive names

MarkAlperin committed:
change conditional styles functionality to use styled-components

MarkAlperin committed:
removed redundant css line

MarkAlperin committed:
destructured props for these files

MarkAlperin committed:
destructure props

MarkAlperin committed:
removed code from pages/Item.js

MarkAlperin committed:
Merge branch 'dev' into mta_browse_page_mvp_functionality

MarkAlperin committed:
fix merge conflict

MarkAlperin committed:
connecting components to global state

MarkAlperin committed:
moved browse directory into the components directory

MarkAlperin committed:
implement View button functionality on item cards

MarkAlperin committed:
merge from changes made on github
Merge branch 'mta_browse_page_mvp_functionality' of https://github.com/GreeneOcean/Project_Greene into mta_browse_page_mvp_functionality

MarkAlperin committed:
delete item/ from this branch

MarkAlperin committed:
added google_api_key to config.js

MarkAlperin committed:
adding HomeItemsWidget.js

MarkAlperin committed:
resolve merge conflicts
Merge branch 'dev' of https://github.com/GreeneOcean/Project_Greene into mta_home_page_items_widget

MarkAlperin committed:
fixed merge conflict

MarkAlperin committed:
picture display added to item page

MarkAlperin committed:
implementd text display

MarkAlperin committed:
map on the page, searching for bug

MarkAlperin committed:
map implemented

MarkAlperin committed:
fixed a map render bug

MarkAlperin committed:
fixed merge conflits
Merge branch 'dev' of https://github.com/GreeneOcean/Project_Greene into mta_item_page

MarkAlperin committed:
fixed port change bug

Doobss commented on client/src/components/item/TextDisplay.js:
```diff
@@ -0,0 +1,88 @@
+
+  const claimHandler = () => {
+    setClaimed(true);
+    //todo: probably a post request
```
```
// todo: probably a post request 
```

@MarkAlperin This is now implemented in the most recent dev branch. So we should get together to set this up.

MarkAlperin replied:
It's... ALIIIVE!

Doobss reviewed (approved):
I'm going to close your Item widget PR from earlier because it seems be redundant considering this one. But again nice work Mark.

Doobss commented on client/src/pages/Item.js:
```diff
@@ -23,10 +24,30 @@ function Item({ state, dispatch, init }) {
 
 
```
@MarkAlperin I'm sure you saw my earlier comment but just to be through, please confirm that both the Home.js and Item.js are matching the dev branch files. (And then add your changes of course lol) thanks @MarkAlperin

MarkAlperin replied:
double checked, looks good

Doobss reviewed (changes requested):
I guess I will request that you confirm Item.js and Home.js will not add any removed code to the dev branch. Thanks @MarkAlperin

7socks commented on client/src/api/index.js:
```diff
@@ -1,7 +1,7 @@
 
-const BASE_URL = `http://localhost:3000`
+const BASE_URL = `http://localhost:8080`
 
```
Is there a reason for this change?

7socks commented on client/src/components/item/ItemPageMapMarker.js:
```diff
@@ -0,0 +1,20 @@
+
+const ItemPageMapMarker = ({ item, selectedItem, $hover, setSelectedItem }) => {
+  const style = { color: "blue", height: "25px", width: "25px" };
+
```
You could use styled component here instead

MarkAlperin replied:
changed

7socks commented on client/src/components/item/ItemPagePhoto.js:
```diff
@@ -0,0 +1,31 @@
+
+const photoNotFoundURL =
+  "https://wellesleysocietyofartists.org/wp-content/uploads/2015/11/image-not-found.jpg";
+
```
Thank you for implementing a default placeholder image! Might want something more generic because it is possible to make a donation without uploading a photo (in that case there is no image to be found anyway) :)

MarkAlperin replied:
more default image, implemented

MarkAlperin committed:
fixed merge conflicts from dev
Merge branch 'dev' of https://github.com/GreeneOcean/Project_Greene into mta_item_page

MarkAlperin committed:
put route functionality for claim button

MarkAlperin committed:
remove console.log

MarkAlperin committed:
changed map marked style to styled-components

MarkAlperin committed:
implement default product image for either pictures: null or pictures: []

MarkAlperin commented:
changes complete
